src/linter/ui5Types/asyncComponentFlags.ts

```diff
@@ -118,38 +117,58 @@ function findAsyncInterface({classDefinition, manifestContent, checker}: {
+		return mergeAsyncFlags(acc ?? {...returnTypeTemplate}, curAnalysis ?? {...returnTypeTemplate});
 	});
 }
 
-function doAsyncInterfaceChecks(importDeclaration: ts.Node): AsyncInterfaceStatus {
-	while (importDeclaration && importDeclaration.kind !== ts.SyntaxKind.SourceFile) {
-		importDeclaration = importDeclaration.parent;
-	}
+function isCoreImportDeclaration(statement: ts.Node): statement is ts.ImportDeclaration {
+	return ts.isImportDeclaration(statement) &&
+		ts.isStringLiteral(statement.moduleSpecifier) &&
+		statement.moduleSpecifier.text === "sap/ui/core/library";
+}
 
-	// @ts-expect-error imports is part of SourceFileObject
-	const moduleImports = importDeclaration?.imports as ts.Node[];
-	const coreLib = moduleImports?.find((importModule) => {
-		return importModule.getText() === "\"sap/ui/core/library\"";
-	}) as ts.StringLiteral | undefined;
-
-	let hasAsyncInterface = AsyncInterfaceStatus.propNotSet;
-	if (coreLib && ts.isImportDeclaration(coreLib.parent)) {
-		if (coreLib.parent.importClause?.namedBindings &&
-			coreLib.parent.importClause.namedBindings.kind === ts.SyntaxKind.NamedImports) {
-			const hasAsyncImport = coreLib.parent.importClause.namedBindings.elements.some(
-				(namedImport) => namedImport.getText() === "IAsyncContentCreation");
+function doAsyncInterfaceChecks(importDeclaration: ts.Node): boolean {
+	const sourceFile = importDeclaration.getSourceFile();
 
-			hasAsyncInterface = hasAsyncImport ? AsyncInterfaceStatus.true : AsyncInterfaceStatus.false;
-		} else {
-			const implementsAsyncInterface =
-				coreLib.parent.importClause?.name?.getText() === "IAsyncContentCreation";
+	let coreLibImports: ts.ImportDeclaration[] | undefined;
+	if (sourceFile.isDeclarationFile) {
+		let moduleDeclaration: ts.ModuleDeclaration | undefined;
+		while (!moduleDeclaration && importDeclaration.kind !== ts.SyntaxKind.SourceFile) {
+			if (ts.isModuleDeclaration(importDeclaration)) {
+				moduleDeclaration = importDeclaration;
+			} else {
+				importDeclaration = importDeclaration.parent;
+			}
+		}
 
-			hasAsyncInterface = implementsAsyncInterface ? AsyncInterfaceStatus.true : AsyncInterfaceStatus.false;
+		if (moduleDeclaration?.body?.kind === ts.SyntaxKind.ModuleBlock) {
+			coreLibImports = moduleDeclaration.body.statements.filter(isCoreImportDeclaration);
 		}
+	} else {
+		coreLibImports = sourceFile.statements.filter(isCoreImportDeclaration);
 	}
 
-	return hasAsyncInterface;
+	if (!coreLibImports) {
+		return false;
+	}
+	const hasAsyncImport = coreLibImports.some((importDecl) => {
+		const importClause = importDecl.importClause;
+		if (!importClause) {
+			return;
+		}
+		if (!importClause.namedBindings) {
+			// Example: import "sap/ui/core/library"; or import library from "sap/ui/core/library";
+		} else if (ts.isNamedImports(importClause.namedBindings)) {
+			// Example: import { IAsyncContentCreation } from "sap/ui/core/library";
```

Comment:
There's also an edge case here- to rename the import 😅 
`import { IAsyncContentCreation: myRenamedAsyncFlag } from ...`